pkg/controller/worker/machines.go

```diff
@@ -223,16 +233,31 @@ func (w *workerDelegate) generateMachineConfig(ctx context.Context) error {
 	return nil
 }
 
-func (w *workerDelegate) generateWorkerPoolHash(pool extensionsv1alpha1.WorkerPool, serverGroupDependency *api.ServerGroupDependency) (string, error) {
+func (w *workerDelegate) generateWorkerPoolHash(pool extensionsv1alpha1.WorkerPool, serverGroupDependency *api.ServerGroupDependency, workerConfig *api.WorkerConfig) (string, error) {
 	var additionalHashData []string
 
 	// Include the given worker pool dependencies into the hash.
 	if serverGroupDependency != nil {
 		additionalHashData = append(additionalHashData, serverGroupDependency.ID)
 	}
 
+	// include machine labels marked for roll
+	var pairs []string
+	for _, pair := range workerConfig.MachineLabels {
+		if pair.Roll {
+			pairs = append(pairs, pair.Name+"="+pair.Value)
+		}
+	}
+	if len(pairs) > 0 {
+		sort.Strings(pairs)
+		additionalHashData = append(additionalHashData, pairs...)
+	}
+
+	// don't include complete provider config in Hash calculation
+	copy := pool
+	copy.ProviderConfig = nil
```

Comment:
i think the main problem we might have with this, is that current ServerGroup machines will be rolled out (because their providerConfig will change). This is a technical debt however and we might as well try to tackle it now than later. WDYT ?

Reply:
There is no perfect solution. With my last changes, it will always use the old hash code calculation as long as there are no machine labels. If a machine label is added (even if it is not setting `triggerRollingOnUpdate=true`), the new hash code calculation will be used and it causes a rolling of the machines as the hash changes. Later changes of not rolling machine labels will not change the hash.
In this way, no rolling happens for existing clusters, but we have the complication of two different hash code calculation. WDYT?